src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -76,6 +78,126 @@ def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right):
             yield {"is_last": is_last, "stride": stride, **processed}
 
 
+class SuffixTrie:
+    """
+    SuffixTrie in Python. Creates a Trie out of a list of objects. The trie is used to find the longest common sequence
+    and return the index in the compared sequence. This is used to merge two chunks of tokens into one.
+    """
+
+    def __init__(self, list=None):
+        self.data = {}
+        sub_lists = [list[i:] for i in range(len(list))]
+        for curr_list in sub_lists:
+            ref = self.data
+            for item in curr_list:
+                ref[item] = item in ref and ref[item] or {}
+                ref = ref[item]
+
+    def add(self, data):
+
+        if not data:
+            # Prevent empty string
+            return
+        ref = self.data
+        for char in data:
+            ref[char] = char in ref and ref[char] or {}
+            ref = ref[char]
+        ref[""] = 1
+
+    def search(self, data):
+        ref = self.data
+        res = []
+        for char in data:
+            if char not in ref:
+                return res
+            start_idx = list(self.data.keys()).index(char)
+            res += [(start_idx, char)]
+            ref = ref[char]
+        return res
+
+
+def _find_timestamp_sequence(sequences, tokenizer, feature_extractor, max_source_positions):
+    """
+    Computes the final sequences by merging the end of the nth sequence with the beginning of the n+1th sequence. Since
+    `WhisperForConditionalGeneration` produces the timestamps pairwise, we filter the consecutive timestamps and only
+    itereate over them. We keep track of the `time` which indicates the actual starting time of the chunk that is
+    processed. We need to make sure to offset the timestamps tokens by the `time` in order for the tokenizer to
+    properly compute the final `offset`.
+    """
+    timestamp_begin = tokenizer.convert_tokens_to_ids("<|notimestamps|>") + 1
+    begin_idx = np.where(sequences[0][0] == timestamp_begin)[1].item()
```

Comment:
This will fail if the begin timestamp token is not in there.

I triggered it with :

```python
from transformers import pipeline
from datasets import load_dataset
import numpy as np

ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
array = ds[40]["audio"]["array"]
pipe = pipeline(
    model="openai/whisper-tiny",
    forced_bos_token_id=None,
    max_new_tokens=448,
    return_timestamps=True,
    chunk_length_s=30,
)

output = pipe(array)
print(output)
```

And there **are** timestamp tokens, just not the begin one.

Reply:
Issue was with the `forced_decoder_ids`  (it's not `forced_bos_token_id` )